### PR TITLE
Resource handling

### PR DESCRIPTION
**Pull request by datazuul**
changed model to be usable for file uploads and removed Resource inheritance. now all "resources" are just identifiables.

### CODE REVIEW

File: dc-model-jackson/src/main/java/de/digitalcollections/model/jackson/DigitalCollectionsModelModule.java [hunk 1]
```diff
@@ -46,41 +55,54 @@
...
+  }
+
   @Override
   public String getModuleName() {
     return "DigitalCollections Model jackson module";
   }
 
+  @Override
+  public Version version() {
+    return VersionUtil.parseVersion("2.0.0-SNAPSHOT", "de.digitalcollections.model", "dc-model-jackson");
```
Comment (tomzirn) [inline dc-model-jackson/src/main/java/de/digitalcollections/model/jackson/DigitalCollectionsModelModule.java]: Vielleicht besser pom parsen? Fehlerquelle ...

File: dc-model-jackson/pom.xml [hunk 2]
```diff
@@ -22,6 +22,7 @@
     <version.junit-jupiter-api>5.2.0</version.junit-jupiter-api>
     <version.junit-jupiter-engine>5.2.0</version.junit-jupiter-engine>
     <version.junit-platform-surefire-provider>1.2.0</version.junit-platform-surefire-provider>
+    <version.log4j>1.2.17</version.log4j>
```
Comment (tomzirn) [inline dc-model-jackson/pom.xml]: logback?!

File: dc-model-jackson/pom.xml [hunk 3]
```diff
@@ -49,6 +50,12 @@
       <artifactId>dc-model</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>log4j</groupId>
```
Comment (tomzirn) [inline dc-model-jackson/pom.xml]: s.o. - stattdessen besser logback?!

File: dc-model-jackson/src/test/java/de/digitalcollections/model/jackson/identifiable/resource/FileResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Test
+  public void testSerialisationInBothWays() throws Exception {
+    checkSerializeDeserialize(createObject());
+  }
+
+  public FileResource createObject() {
+    FileResource obj = new FileResourceImpl();
+    obj.setFilename("filename.jpg");
+    obj.setFilenameExtension("jpg");
```
Comment (tomzirn) [inline dc-model-jackson/src/test/java/de/digitalcollections/model/jackson/identifiable/resource/FileResourceTest.java]: braucht's das hier nochmal? eh schon teil des dateinamens ...

File: dc-model/pom.xml [hunk 5]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>de.digitalcollections.model</groupId>
     <artifactId>dc-model-parent</artifactId>
-    <version>1.1.0-SNAPSHOT</version>
+    <version>2.0.0-SNAPSHOT</version>
```
Comment (tomzirn) [inline dc-model/pom.xml]: hier warnung "ist hardgecoded im jackson-module" o.ä.?

File: dc-model-jackson/pom.xml (hunk 2 above)
Comment (datazuul) [reply dc-model-jackson/pom.xml]: introduced logback, see https://github.com/dbmdz/digitalcollections-model/commit/71f259b1d0c1a3316d03ef776074d080c60606fc

File: dc-model-jackson/pom.xml (hunk 3 above)
Comment (datazuul) [reply dc-model-jackson/pom.xml]: introduced logback, see https://github.com/dbmdz/digitalcollections-model/commit/71f259b1d0c1a3316d03ef776074d080c60606fc

File: dc-model/pom.xml (hunk 5 above)
Comment (datazuul) [reply dc-model/pom.xml]: Externalized to properties file: solved by https://github.com/dbmdz/digitalcollections-model/commit/94da371b44ca7f38888784a4fd6e6398d0eee811

**Comment by datazuul** [comment]
all comments fixed

**Pull request merged**